product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md

```diff
@@ -0,0 +1,178 @@
+</ser:request>
+<xsd:symbol>SUN</xsd:symbol>
+</ser:getQuote>
+</soapenv:Body>
+</soapenv:Envelope>
+```
+
+This slight modification to the payload can be done using enrich mediator. 
+
+mediator
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<enrich xmlns="http://ws.apache.org/ns/synapse">
+    <source clone="true" type="inline">
+        <xsd:symbol xmlns:xsd="http://services.samples/xsd">SUN</xsd:symbol>
+    </source>
+    <target action="sibling" type="custom"
+        xmlns:ser="http://services.samples"
+        xmlns:xsd="http://services.samples/xsd" xpath="//ser:getQuote/ser:request"/>
+</enrich>
+```
+
+## Prerequisites
+SimpleStockQuote Service needs to have deployed. 
+
+## Development guidelines
+
+Sequence and proxy Configuration
+
+Create the following sequence and add to the proxy service. 
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy xmlns="http://ws.apache.org/ns/synapse"
+       name="Enrich3"
+       startOnLoad="true"
+       statistics="disable"
+       trace="disable"
+       transports="http,https">
+   <target>
+      <inSequence>
+         <log level="full"/>
+         <enrich>
+            <source clone="true" type="inline">
+               <xsd:symbol xmlns:xsd="http://services.samples/xsd">SUN</xsd:symbol>
+            </source>
+            <target xmlns:xsd="http://services.samples/xsd"
+                    xmlns:ser="http://services.samples"
+                    action="sibling"
+                    xpath="//ser:getQuote/ser:request"/>
+         </enrich>
+         <log level="full"/>
+      </inSequence>
+      <endpoint>
+         <address uri="http://localhost:9000/services/SimpleStockQuoteService"/>
+      </endpoint>
+   </target>
+   <description/>
+</proxy>
+                                
+```
+
+Invoke the getQuote service with the following payload. Use an SOAP client like SOAP UI. 
+Proxy: http://localhost:8280/services/Enrich2
+
+```
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ser="http://services.samples" xmlns:xsd="http://services.samples/xsd">
+<soapenv:Header/>
+<soapenv:Body>
+<ser:getQuote>
+<!--Optional:-->
+<ser:request>
+<!--Optional:-->
+<xsd:symbol>WSO2</xsd:symbol>
+</ser:request>
+</ser:getQuote>
+</soapenv:Body>
+</soapenv:Envelope>
+```
+
+Above proxy will add the sibling element <xsd:symbol>SUN</xsd:symbol> to the <ser:request> element before the payload is sent to the endpoint. Below is the log when the proxy is called. You can see the sibling element has been added after the Enrich3 proxy is called.  
+
+
+```
+[2018-11-26 15:39:23,302] [EI-Core]  INFO - LogMediator To: /services/Enrich3, WSAction: urn:getQuote, SOAPAction: urn:getQuote, MessageID: urn:uuid:40ae063a-9eae-487c-afde-052eb2d2e940, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:xsd="http://services.samples/xsd" xmlns:ser="http://services.samples"><soap:Body>
```

Comment:
Let's remove debug logs.
Let's try to demonstrate use cases by responding to the client

Reply:
Removed all the logs.